horace_core/sqw/page_operations/PageOp_split_sqw.m

```diff
@@ -0,0 +1,214 @@
+classdef PageOp_split_sqw < PageOpBase
+    % Single page pixel operation and main gateway for
+    % sqw.split  algorithm.
+    %
+    %
+    properties
+        % cellarray of the images for resulting sqw objects to be split into
+        out_img
+        % cellarray of the pixels for resulting sqw objects to be split into
+        out_pix;
+        % cellarray of target sqw objects to split original
+        % Needed for correct write handle initialization
+        out_sqw;
+        % cellarray of pixel ranges for all contributing objects
+        obj_pix_ranges;
+        % cell array of write handles, used to store data for target sqw
+        % files.
+        write_handles;
+    end
+    properties(Access = private)
+        % the runid map for the initial sqw object to split
+        runid_map_;
+        % numbers of objects contributing to current page
+        n_obj_contrib_to_page_;
+        % array containing true if run descriped in experiment contributest
+        % to pixels and false if it does not
+        run_contributes_
+    end
+    methods
+        function obj = PageOp_split_sqw(varargin)
+            obj = obj@PageOpBase(varargin{:});
+            obj.op_name_ = 'split_sqw';
+            obj.split_at_bin_edges = false;
+        end
+
+        function obj = init(obj,in_sqw)
+            % indirect usage of inplace property. Use by-product of
+            % inplace == true option when if it is true, write handle is
+            % not generated for in_sqw as we do not need write handle for
+            % init_sqw
+            obj.inplace = true;
+            obj = init@PageOpBase(obj,in_sqw);
+            % if write handle was generated for in_sqw due to errors in a
+            % future, clear it up. We do not need it here
+            obj.write_handle_ = [];
+            % we do not need image for this type of operations. Save the
+            % memory.
+            obj.img_ = [];
+            % prepare target sqw objects to split source into
+            obj = obj.prepare_split_sqw(in_sqw);
+            % initialize target pix ranges for evaluation
+            n_objects = numel(obj.out_img);
+            obj.obj_pix_ranges = arrayfun(@(nobj)PixelDataBase.EMPTY_RANGE(), ...
+                1:n_objects, 'UniformOutput',false);
```

Comment:
Is this equivalent to
```suggestion
            obj.obj_pix_ranges = repmat(PixelDataBase.EMPTY_RANGE(), [1, 1, n_objects]);
```
At any rate, `EMPTY_RANGE` is a constant, not a function, doesn't need parens.

Reply:
no -- suggested operation chreated array and above builds cellarray.

Reply:
In which case
```suggestion
            obj.obj_pix_ranges = repmat({PixelDataBase.EMPTY_RANGE}, [1, n_objects])
```

Reply:
well, this is probably OK, but why it is better then what is already there?

Reply:
Because it states the intent explicitly. You don't want to apply a function here. You want to create an array which is a `rep`eat of the `mat`rix. This makes it self documenting and removes some of the obscurity.

Reply:
I may say that I am explicitly initialize the state for cellarray of page operations.